File: CF/stresstestwithchecker.py
from os import WEXITSTATUS, system
import random


file = './1364C'
inp = './in'
outp = './out'


def check(arr: 'list[str]'):
    status = system(file)
    if WEXITSTATUS(status) != 0:
        print('RTE')
        exit(1)
    foutput = open(outp)
    ans = [int(x) for x in foutput.readline().split(' ')]
    if ans[0] == -1:
        # print("Accepted")
        return
    mxx = [x for x in range(0, 1000001)]
    for i in range(0, len(arr)):
        if ans[i] in mxx:
            mxx.remove(ans[i])
        if int(arr[i]) != mxx[0]:
            print('Wrong Answer')
            exit(1)
    # print("Accepted")


def generate_tc(n: 'int', arr: 'list[str]'):
    if len(arr) == n:
        finput = open(inp, 'w')
        finput.write(f'{str(n)}\n')
        finput.write(' '.join(arr) + '\n')
        finput.close()
        check(arr)
        # exit(0)
        return
    for i in range(1, 11):
        arr.append(str(i))
        generate_tc(n, arr)
        arr.pop()


n = 5
generate_tc(n, [])
# inp_file = open(inp, 'w')
# inp_file.write(f'{len(arr)}\n')
# for i in range(0, len(arr)):
#     inp_file.write(f'{str(n)}\n{arr[i]}\n')

# inp_file.close()
# compare()

# print(arr)

File: CF/stresstestwithchecker.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;
typedef vector<ll> vl;

#define CMD_TO_RUN "./1364C"
#define INPUT_FILE "./in"
#define OUTPUT_FILE "./out"

inline void check(vl &arr) {
    int status = system(CMD_TO_RUN);
    if (status != 0) {
        printf("Runtime Error\n");
        exit(0);
    }
    FILE *output = fopen(OUTPUT_FILE, "r");
    if (!output) {
        printf("Error while reading file\n");
        exit(1);
    }
    vl ans;
    ll tmp;
    while(fscanf(output, "%lld", &tmp) != EOF) {
        ans.push_back(tmp);
    }
    if(ans.size() == 0U) {
        printf("Wrong Answer\n");
        exit(0);
    }
    if(ans.size() == 1 && ans[0] == -1) {
        printf("Skipped\n");
        return;
    }
    set<int> mex;
    for(int i=0; i<100; i++) {
        mex.insert(i);
    }
    for(int i=0; i<(int) ans.size(); i++) {
        auto fnd = mex.find(ans[i]);
        if(fnd != mex.end())
            mex.erase(fnd);
        if(arr[i] != *mex.begin() || ans[i] > 1e6 || ans[i] < 0) {
            printf("Wrong Answer\n");
            exit(0);
        }
    }
    printf("Accepted\n");
    fclose(output);
}

void generateTC(int n, vl &arr) {
    if ((int)arr.size() == n) {
        FILE *input = fopen(INPUT_FILE, "w");
        if (!input) {
            printf("Error while reading file\n");
            exit(1);
        }
        fprintf(input, "%d\n", n);
        for(int i=0; i<n; i++)
            fprintf(input, "%lld%c", arr[i], " \n"[i == n - 1]);
        fclose(input);
        check(arr);
        return;
    }
    for (int i=(arr.empty() ? 0 : arr.back()); i<11; i++) {
        arr.push_back(i);
        generateTC(n, arr);
        arr.pop_back();
    } 
}

int main() {
    int n = 5;
    vl arr;
    generateTC(n, arr);
    return 0;
}